stream_read.py

```diff
@@ -0,0 +1,484 @@
+                        # creat list
+                        cstar = [float(x) for x in line.split(' ')[2:-1]]
+                    elif "lattice_type" in line:
+                        # I found a line `lattice_type`
+                        flags["lattice_type"] = True
+                        lattice_type = line.strip().split(' ')[2]
+                    elif "centering" in line:
+                        # I found a line `centering`
+                        flags["centering"] = True
+                        centering = line.strip().split(' ')[2]
+                    elif "unique_axis" in line:
+                        # I found a line `unique_axis`
+                        flags["unique_axis"] = True
+                        unique_axis = line.strip().split(' ')[2]
+
+                if "--- End crystal" in line:
+                        # the end line of cryst. info.
+                    # We need `astar`, `bstar` and `cstar`
+                    # to calculate unit cell parameters
+                    if not(flags["astar"] or flags["bstar"] or flags["cstar"]):
+                        print("Image {} has bad cell".format(name))
+                    else:
+                        a, b, c, alfa, beta, gamma =\
+                            cell_parameters(astar, bstar, cstar)
+                        if not (flags["lattice_type"] and
+                                flags["centering"] and flags["unique_axis"]):
+
+                            # if I do not have `lattice_type` ,`centering`
+                            # or `unique_axis`then
+                            # I keep the default (`triclinic` `P` , `?`)
+                            print("{} keep default triclinic P".format(name))
+                            lattice_type = "triclinic"
+                            centering = "P"
+                            unique_axis = "?"
+
+                        crystal = {'name': name, 'a': a, 'b': b, 'c': c,
+                                   'alfa': alfa, 'beta': beta,
+                                   'gamma': gamma, 'centering': centering,
+                                   'lattice_type': lattice_type,
+                                   'unique_axis': unique_axis}
+                        crystals.append(crystal)
+                        # reset event_name
+                        event_name = ""
+                        # reset flags
+                        for key in flags.keys():
+                            flags[key] = False
+    except TypeError:
+        print("Wrong path to the stream file.")
+        sys.exit()
+    except IndexError:
+        print("Enter path to the stream file.")
+        sys.exit()
+    except FileNotFoundError:
+        print("File not found or not a indexing stream file.")
+        sys.exit()
+    print("Loaded {} cells from {} chunks".format(len(crystals),
+                                                  chunks_counter))
+    return crystals
+
+
+class PeakSearch:
```

Comment:
Note for later: A more pythonic way of doing this is using a data class: https://stackoverflow.com/questions/47955263/what-are-data-classes-and-how-are-they-different-from-common-classes

This comment applies to any of the classes where you say "Class for representing..."

Reply:
that's nice but I think data classes only appear in python `3.7`, we aim at supporting at least from `3.5`

Reply:
I changed Classes ("Class for representing..") to dictionaries in commit b903e317f6eea8f42c1a1dc4748d11769b4c68bc Is this acceptable?

Reply:
Yep, that's a decent way to do it